addons/binding/org.openhab.binding.yamahareceiver/src/main/java/org/openhab/binding/yamahareceiver/internal/protocol/xml/InputWithPlayControlXML.java

```diff
@@ -0,0 +1,196 @@
+        }
+
+        PlayInfoState msg = new PlayInfoState();
+
+        Node playbackInfoNode = XMLUtils.getNode(doc.getFirstChild(), "Play_Info/Playback_Info");
+        msg.playbackMode = playbackInfoNode != null ? playbackInfoNode.getTextContent() : "";
+
+        Node metaInfoNode = XMLUtils.getNode(doc.getFirstChild(), "Play_Info/Meta_Info");
+        Node sub;
+
+        if (inputID.equals("TUNER")) {
+            sub = XMLUtils.getNode(metaInfoNode, "Radio_Text_A");
+            msg.station = sub != null ? sub.getTextContent() : "";
+        } else {
+            sub = XMLUtils.getNode(metaInfoNode, "Station");
+            msg.station = sub != null ? sub.getTextContent() : "";
+        }
+
+        sub = XMLUtils.getNode(metaInfoNode, "Artist");
+        msg.artist = sub != null ? sub.getTextContent() : "";
+
+        sub = XMLUtils.getNode(metaInfoNode, "Album");
+        msg.album = sub != null ? sub.getTextContent() : "";
+
+        sub = XMLUtils.getNode(metaInfoNode, "Song");
+        msg.song = sub != null ? sub.getTextContent() : "";
+
+        observer.playInfoUpdated(msg);
+    }
+
+    /**
+     * Start the playback of the content which is usually selected by the means of the Navigation control class or
+     * which has been stopped by stop().
+     *
+     * @throws Exception
+     */
+    @Override
+    public void play() throws IOException, ReceivedMessageParseException {
+        comReference.get().send(wrInput("<Play_Control><Playback>Play</Playback></Play_Control>"));
+        update();
+    }
+
+    /**
+     * Stop the currently playing content. Use start() to start again.
+     *
+     * @throws Exception
+     */
+    @Override
+    public void stop() throws IOException, ReceivedMessageParseException {
+        comReference.get().send(wrInput("<Play_Control><Playback>Stop</Playback></Play_Control>"));
+        update();
+    }
+
+    /**
+     * Pause the currently playing content. This is not available for streaming content like on NET_RADIO.
+     *
+     * @throws Exception
+     */
+    @Override
+    public void pause() throws IOException, ReceivedMessageParseException {
+        comReference.get().send(wrInput("<Play_Control><Playback>Pause</Playback></Play_Control>"));
+        update();
+    }
+
+    /**
+     * Skip forward. This is not available for streaming content like on NET_RADIO.
+     *
+     * @throws Exception
+     */
+    @Override
+    public void skipFF() throws IOException, ReceivedMessageParseException {
+        comReference.get().send(wrInput("<Play_Control><Playback>Skip Fwd</Playback></Play_Control>"));
+    }
+
+    /**
+     * Skip reverse. This is not available for streaming content like on NET_RADIO.
+     *
+     * @throws Exception
+     */
+    @Override
+    public void skipREV() throws IOException, ReceivedMessageParseException {
+        comReference.get().send(wrInput("<Play_Control><Playback>Skip Rev</Playback></Play_Control>"));
```

Comment:
@martinvw I actually refactored some of this in my other branch ([2574_yamaha_spotify](https://github.com/zarusz/openhab2-addons/tree/2574_yamaha_spotify)) that will add Spotify feature #2574. I intended to merge it to this PR. Do you suggest I merge to this PR or you could wait with this until another PR ?

I also wondered about update, and decided to add this. Yet there will be just one method that eliminates the boilerplate code.